integration-tests/tests/src/tests/sync/flexible.ts

```diff
@@ -396,6 +399,86 @@ describe.skipIf(environment.missingServer, "Flexible sync", function () {
     });
   });
 
+  describe("Errors", () => {
+    it("raises compensating writes errors", async function () {
+      const person1Id = new BSON.ObjectId("0000002a9a7969d24bea4cf5");
+      const person2Id = new BSON.ObjectId("0000002a9a7969d24bea4cf6");
+      const dogId = new BSON.ObjectId("0000002a9a7969d24bea4cf7");
+
+      await new Promise<void>((resolve, _) => {
```

Comment:
Does this even need to be wrapped in a promise?  We could just await the realm.open and the upload.  Mocha has a `done()` argument to the test method that can be included in the error callback.

Reply:
When the upload is finished it's not guaranteed that the error is raised, that's why I did it like this.

Thinking about it... I can maybe add a wait for the download after the upload, and maybe that would work.

Reply:
https://mochajs.org/#asynchronous-code

Reply:
Thanks for the link! That's probably what I need

Reply:
I actually learned about the `done()` argument when reviewing the test migrations.  Very cool feature for async code.

Reply:
It seems I can't use this, because of `Error: Resolution method is overspecified. Specify a callback *or* return a Promise; not both.`
It seems because the body of the method is async, then it's implicitly a promise.

Reply:
Ok, was worth a shot.